### PR TITLE
Fix Linux devcontainer issue from file permissions

### LINKED ISSUE #171: VS Code Devcontainer fails to set up on Linux
### Expected Result
The devcontainer is successfully set up.

### Actual Result
An error is thrown when trying to run the `postAttachCommand`, specifically when we're trying to install Cypress.

### Steps to Reproduce
1. Have Linux as your host operating system
2. Make sure you've built the base Docker image `benefits_client:latest`
3. Open the `benefits` folder as a remote container in VS Code

### Output
From console logging:
```
Running the postAttachCommand from devcontainer.json...

[117148 ms] Start: Run in container: /bin/bash localhost/bin/init.sh
[117388 ms] Port forwarding connection from 59748 > 37657 > 37657 in the container.
[117389 ms] Start: Run in container: /root/.vscode-server/bin/6cba118ac49a1b88332f312a8f67186f7f3c1643/node -e 
pre-commit installed at .git/hooks/pre-commit
[WARNING] The 'rev' field of repo 'https://github.com/compilerla/conventional-pre-commit' appears to be a mutable reference (moving tag / branch).  Mutable references are never updated after first install and are not supported.  See https://pre-commit.com/#using-the-latest-version-for-a-repository for more details.  Hint: `pre-commit autoupdate` often fixes this.
[INFO] Initializing environment for https://github.com/compilerla/conventional-pre-commit.
...
[INFO] Initializing environment for https://github.com/pre-commit/pre-commit-hooks.
[119288 ms] [15:54:45] Extracted extension to /root/.vscode-server/extensions/.c2d52185-63c1-4349-ac91-40616d2fdbc8: ms-python.python
[119306 ms] [15:54:45] Renamed to /root/.vscode-server/extensions/ms-python.python-2021.10.1365161279
[119310 ms] [15:54:45] Extracting completed. ms-python.python
[119311 ms] [15:54:45] Extension installed successfully: ms-toolsai.jupyter-keymap
[119311 ms] [15:54:45] Extension installed successfully: ms-toolsai.jupyter-renderers
[119311 ms] [15:54:45] Extension installed successfully: ms-toolsai.jupyter
[15:54:45] Extension installed successfully: ms-python.python
[119316 ms] Extension 'ms-python.python' v2021.10.1365161279 was successfully installed.
[INFO] Initializing environment for https://github.com/psf/black.
[INFO] Initializing environment for https://gitlab.com/pycqa/flake8.
[120807 ms] [15:54:46] Extracted extension to /root/.vscode-server/extensions/.1a0abef0-8ca2-4923-8aa5-ec1605a86354: ms-python.vscode-pylance
[120821 ms] [15:54:46] Renamed to /root/.vscode-server/extensions/ms-python.vscode-pylance-2021.10.3
[120823 ms] [15:54:46] Extracting completed. ms-python.vscode-pylance
[120823 ms] [15:54:46] Extension installed successfully: ms-python.vscode-pylance
[120826 ms] Extension 'ms-python.vscode-pylance' v2021.10.3 was successfully installed.
[121800 ms] Extensions cache, remote removals: None
[INFO] Initializing environment for https://github.com/pycqa/bandit.
[INFO] Installing environment for https://github.com/pre-commit/pre-commit-hooks.
[INFO] Once installed this environment will be reused.
[INFO] This may take a few minutes...
...
[INFO] Installing environment for https://github.com/psf/black.
[INFO] Once installed this environment will be reused.
[INFO] This may take a few minutes...
[INFO] Installing environment for https://gitlab.com/pycqa/flake8.
[INFO] Once installed this environment will be reused.
[INFO] This may take a few minutes...
[INFO] Installing environment for https://github.com/pycqa/bandit.
[INFO] Once installed this environment will be reused.
[INFO] This may take a few minutes...
pre-commit installed at .git/hooks/commit-msg

up to date, audited 171 packages in 765ms

25 packages are looking for funding
  run `npm fund` for details

found 0 vulnerabilities
Cypress cannot write to the cache directory due to file permissions

See discussion and possible solutions at
https://github.com/cypress-io/cypress/issues/1281

----------

Failed to access /root/.cache/Cypress:

EACCES: permission denied, mkdir '/root/.cache/Cypress'

----------

Platform: linux (Debian - 11)
Cypress Version: 8.7.0
```
### Notes
@thekaveman and I speculate it has something to do with the "quirks with local filesystem (bind) mounts" with Docker CE/EE on Linux (see [VS Code docs on non-root users](https://code.visualstudio.com/remote/advancedcontainers/add-nonroot-user)). Windows users can see everything in the Linux container because there isn't a way to map the permissions. [This comment](https://github.com/microsoft/vscode-remote-release/issues/1155#issuecomment-521077955) may have some helpful details too.

There are other GitHub issues showing this error:
 - https://github.com/cypress-io/cypress/issues/15025
 - https://github.com/cypress-io/cypress/issues/8093

### PR DESCRIPTION
**Pull request by angela-tran**
Closes #171 

- Added lines to `Dockerfile.dev` to use the `calitp` user so that the npm dependencies are installed using that user rather than root
- Added lines in `Dockerfile` to make the Python packages be installed using `calitp` as well

### CODE REVIEW

File: localhost/Dockerfile.dev [hunk 1]
```diff
@@ -13,5 +13,8 @@ RUN curl -fsSL https://deb.nodesource.com/setup_16.x | bash
 RUN apt-get install -qq nodejs libgtk2.0-0 libgtk-3-0 libgbm-dev libnotify-dev \
                         libgconf-2-4 libnss3 libxss1 libasound2 libxtst6 xauth xvfb
 
+# todo: need to extract this
+USER calitp
```
Comment (angela-tran) [inline localhost/Dockerfile.dev:17]: Oops didn't mean to leave the todo in... but the point still stands that I want to use `$USER` if that is still visible from `Dockerfile`. Need to test if that works. Or chime in if you know that it works!

Comment (angela-tran) [reply localhost/Dockerfile.dev]: > The environment variables set using ENV will persist when a container is run from the resulting image. 

From https://docs.docker.com/engine/reference/builder/#env

**Review by thekaveman — COMMENTED** [review]
@angela-tran this is great! Your investigation has uncovered that we can do even less as the `root` user, including running the devcontainer as non-root `calitp` user! I made a few minor reorganizations and changes (it was easier than typing out a review / trying to explain in writing 😅).

Can you please verify it still in a Linux environment with these changes? Make sure you start from scratch:

* On your host machine, `docker-compose build --no-cache client`
* Rebuild and Reopen the devcontainer
  * Can you start the app locally with `F5`?
  * Can you run through the eligibility flow locally?
  * Can you run `cd tests/e2e && npx cypress run`?

**Comment by thekaveman** [comment]
@machikoyasuda tagging you in for a review as well. Can you please take a look at the issue #171 and try to run everything from this branch, to make sure it all still works on Mac?

**Comment by machikoyasuda** [comment]
I ran through these steps and was able to run all of them successfully:
1. On your host machine, `docker-compose build --no-cache client` ✅
2. Rebuild and Reopen the devcontainer ✅
3. Can you start the app locally with F5? ✅
4. Can you run through the eligibility flow locally? ✅
5. Can you run `cd tests/e2e && npx cypress run`? ✅

image
image
image
image

**Review by machikoyasuda — APPROVED** [review]

**Comment by angela-tran** [comment]
@thekaveman, your changes look good to me! In fact, I think your change to install pip with the non-root user fixed an issue I started seeing where VS Code couldn't see pip.

_(It would complain that flake8 wasn't installed and then say that it couldn't find pip.)_
image
image

I successfully ran through the steps below that are marked with a checkmark :

> * On your host machine, `docker-compose build --no-cache client` :heavy_check_mark: 
> * Rebuild and Reopen the devcontainer :heavy_check_mark: 
>   * Can you start the app locally with `F5`? :heavy_check_mark: 
>   * Can you run through the eligibility flow locally? 
>   * Can you run `cd tests/e2e && npx cypress run`? :heavy_check_mark: 

I can run through most of the flow but am not set up to get to the very end (need to set up eligibility-server). I don't see any problems currently.

Here are the successful cypress tests:
image
image

**Pull request merged**